File: 0100-same-tree/0100-same-tree.cpp
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        return traverse(p,q);
    }

    bool traverse(TreeNode* p, TreeNode* q){
        if(p == nullptr && q == nullptr) return true;
        if(p == nullptr || q == nullptr) return false;
        if(p->val != q->val)             return false;
        return traverse(p->left,q->left) && traverse(p->right,q->right);  
    }
};

File: 0100-same-tree/0100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        answer = [True]
        self.recursion(p,q,answer)
        return answer[0]

    def recursion(self,one,two,answer):
        
        if one == None and two == None:
            return 

        if one == None and two != None:
            answer[0] = False 
            return
            
        if one != None and two == None:
            answer[0] = False 
            return
            
        if one.val != two.val:
            answer[0] = False
        
        if answer[0] == False:
            return False
            
        self.recursion(one.left,two.left,answer)
        self.recursion(one.right,two.right,answer)